### PR TITLE
nixos/profiles/minimal: set fonts.enableDefaultPackages which was previously set by environment.noXLibs

### PR DESCRIPTION
**Pull request by SuperSandro2000**
…ironment.noXLibs

## Description of changes

They are probably worthwhile to keep and I forgot to copy them over in the last PR.

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: nixos/modules/profiles/minimal.nix [hunk 1]
```diff
@@ -21,6 +21,8 @@ with lib;
 
   environment.stub-ld.enable = false;
 
+  fonts.fontconfig.enable = false;
```
Comment (emilazy) [inline nixos/modules/profiles/minimal.nix:24]: Maybe it’s best to just disable installation of default fonts, as the configuration file itself takes up little space?

File: nixos/modules/profiles/minimal.nix [hunk 2]
```diff
@@ -29,6 +31,10 @@ with lib;
 
   programs.command-not-found.enable = mkDefault false;
 
+  programs.ssh.setXAuthLocation = false;
+
+  security.pam.services.su.forwardXAuth = lib.mkForce false;
```
Comment (emilazy) [inline nixos/modules/profiles/minimal.nix:36]: I don’t see any argument for setting these if `noXlibs` is not being used.

**Review by emilazy — CHANGES_REQUESTED** [review]

File: nixos/modules/profiles/minimal.nix (hunk 1 above)
Comment (SuperSandro2000) [reply nixos/modules/profiles/minimal.nix:24]: I don't know how to do this. Looking at the module there seems to be no way to remove the fonts https://github.com/NixOS/nixpkgs/blob/nixos-unstable/nixos/modules/config/fonts/fontconfig.nix#L287-L357

Comment (emilazy) [reply nixos/modules/profiles/minimal.nix:24]: It’s `fonts.enableDefaultPackages`; it’s off by default, and set by e.g. `services.graphical-desktop.enable`. I think we could probably just drop this?

File: nixos/modules/profiles/minimal.nix [hunk 3]
```diff
@@ -29,6 +31,8 @@ with lib;
 
   programs.command-not-found.enable = mkDefault false;
 
+  programs.ssh.setXAuthLocation = false;
```
Comment (emilazy) [inline nixos/modules/profiles/minimal.nix:34]: This should be `mkDefault false` like the other settings.

**Review by emilazy — CHANGES_REQUESTED** [review]

**Review by emilazy — APPROVED** [review]
LGTM, thanks!

**Comment by XYenon** [comment]
This causes an error when `services.openssh.settings.X11Forwarding = true;`

```
       error: The option `programs.ssh.setXAuthLocation' has conflicting definition values:
       - In `/nix/store/xnjw9gmfmpppdj6bxpw6cfkspc3h6xwl-source/nixos/modules/profiles/minimal.nix': false
       - In `/nix/store/xnjw9gmfmpppdj6bxpw6cfkspc3h6xwl-source/nixos/modules/programs/ssh.nix': true
       Use `lib.mkForce value` or `lib.mkDefault value` to change the priority on any of these definitions.
```

**Comment by emilazy** [comment]
`programs.ssh.setXAuthLocation` is set with `lib.mkDefault` in both places. The default in this profile should have its priority adjusted to lose to that one, I guess. Or it could just be removed, since it looks like it’d be `false` in the obvious cases anyway.

**Pull request merged**